Source/Modules/lua.cxx

```diff
@@ -767,9 +773,19 @@ class LUA:public Language {
     Printv(f->code, "return SWIG_arg;\n", NIL);
     // add the failure cleanup code:
     Printv(f->code, "\nif(0) SWIG_fail;\n", NIL);
-    Printv(f->code, "\nfail:\n", NIL);
-    Printv(f->code, "$cleanup", "lua_error(L);\n", NIL);
-    Printv(f->code, "return SWIG_arg;\n", NIL);
+    Printv(f->code, "\nfail:\n", "$cleanup", NIL);
+    if (CPlusPlus) {
+      if (!Strchr(cleanup, ';') && !Strchr(cleanup, '}')) {
+	// If there's no cleanup code we need a dummy statement after the label.
+	Append(f->code, ";\n}\n");
```

Comment:
Actually, I think this extra `;` needs to be conditional only on `CPlusPlus` - otherwise setting the cleanup code to e.g. `// This is a dummy comment; it should still work!` would result in an uncompilable wrapper...